### PR TITLE
sync: support cancel_rx in ttrpc context

### PR DESCRIPTION
**Pull request by yuqitao**
Fixes: https://github.com/containerd/ttrpc-rust/issues/180

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -20,6 +20,7 @@ thiserror = "1.0"
 async-trait = { version = "0.1.31", optional = true }
 tokio = { version = "1", features = ["rt", "sync", "io-util", "macros", "time"], optional = true }
 futures = { version = "0.3", optional = true }
+crossbeam = "0.8.0"
```
[the remark this answers is not in the document]
Comment (yuqitao) [reply Cargo.toml:23]: With crossbeam channel, we can use `select!` in ttrpc method handler.
like: 
```
select! {
       recv(ticker) -> _ => println!("elapsed: {:?}", start.elapsed()),
       recv(timeout) -> _ => break,
}
```

[the remark this answers is not in the document]
Comment (jsturtevant) [reply Cargo.toml:23]: interesting!  seems like a possibly good improvement thought I would tend towards bring in crossbeam as dep in separate PR across all the areas where   std::sync::mpsc::Receiver/Sender is used to keep it similiar throughout the codebase.  I will defer to the maintainers though.

File: src/sync/server.rs [hunk 2]
```diff
@@ -146,22 +139,18 @@ fn start_method_handler_thread(
...
                         quit.store(true, Ordering::SeqCst);
-                        // the client connection would be closed and
+                        // the workload tx would be dropped and
                         // the connection dealing main thread would
                         // have exited.
                         control_tx
                             .send(())
                             .unwrap_or_else(|err| trace!("Failed to send {:?}", err));
-                        trace!("Socket error send control_tx");
+                        trace!("wrokload_rx recv error, send control_tx");
```
[the remark this answers is not in the document]
Comment (yuqitao) [reply src/sync/server.rs]: done

**Comment by yuqitao** [comment]
@lifupan @jsturtevant PTAL.

**Review by lifupan — APPROVED** [review]

**Comment by yuqitao** [comment]
@Tim-Zhang The CI seems to have some wired error, could you help solving it please?

**Review by Tim-Zhang — APPROVED** [review]
LGTM, thanks @yuqitao

**Pull request merged**